packages/twenty-front/src/modules/workflow/components/WorkflowEditActionFormServerlessFunction.tsx

```diff
@@ -53,166 +18,17 @@ export const WorkflowEditActionFormServerlessFunction = ({
-      (f) => f.id === newServerlessFunctionId,
-    );
-
-    const newProps = {
-      ...action,
-      settings: {
-        ...action.settings,
-        input: {
-          serverlessFunctionId: newServerlessFunctionId,
-          serverlessFunctionVersion:
-            serverlessFunction?.latestVersion || 'latest',
-          serverlessFunctionInput: getFunctionInput(newServerlessFunctionId),
-        },
-      },
-    };
+  const { loading: isLoadingServerlessFunctions } =
+    useGetManyServerlessFunctions();
 
-    actionOptions.onActionUpdate(newProps);
-  };
-
-  const renderFields = (
-    functionInput: FunctionInput,
-    path: string[] = [],
-    isRoot = true,
-  ): ReactNode[] => {
-    const displaySeparator = (functionInput: FunctionInput) => {
-      const keys = Object.keys(functionInput);
-      if (keys.length > 1) {
-        return true;
-      }
-      if (keys.length === 1) {
-        const subKeys = Object.keys(functionInput[keys[0]]);
-        return subKeys.length > 0;
-      }
-      return false;
-    };
-
-    return Object.entries(functionInput).map(([inputKey, inputValue]) => {
-      const currentPath = [...path, inputKey];
-      const pathKey = currentPath.join('.');
-
-      if (inputValue !== null && typeof inputValue === 'object') {
-        if (isRoot) {
-          return (
-            <>
-              {displaySeparator(functionInput) && (
-                <HorizontalSeparator noMargin />
-              )}
-              {renderFields(inputValue, currentPath, false)}
-            </>
-          );
-        }
-        return (
-          <StyledContainer key={pathKey}>
-            <StyledLabel>{inputKey}</StyledLabel>
-            <StyledInputContainer>
-              {renderFields(inputValue, currentPath, false)}
-            </StyledInputContainer>
-          </StyledContainer>
-        );
-      } else {
-        return (
-          <VariableTagInput
-            key={pathKey}
-            inputId={`input-${inputKey}`}
-            label={inputKey}
-            placeholder="Enter value"
-            value={`${inputValue || ''}`}
-            onChange={(value) => handleInputChange(value, currentPath)}
-          />
-        );
-      }
-    });
-  };
+  if (isLoadingServerlessFunctions) {
+    return null;
+  }
 
   return (
-    <WorkflowEditGenericFormBase
-      HeaderIcon={<IconCode color={theme.color.orange} />}
-      headerTitle="Code - Serverless Function"
-      headerType="Code"
-    >
-      <Select
-        dropdownId="select-serverless-function-id"
-        label="Function"
-        fullWidth
-        value={action.settings.input.serverlessFunctionId}
-        options={availableFunctions}
-        emptyOption={{ label: 'None', value: '' }}
-        disabled={actionOptions.readonly}
-        onChange={handleFunctionChange}
-      />
-      {renderFields(functionInput)}
-    </WorkflowEditGenericFormBase>
+    <WorkflowEditActionFormServerlessFunctionInner
```

Comment:
why adding another component here? I feel like the parent does not do a lot. Let's keep everything in the same?